delfin/drivers/ibm/storwize_svc/ssh_handler.py

```diff
@@ -0,0 +1,409 @@
+    OID_LAST_TIME = '1.3.6.1.4.1.2.6.190.4.10'
+    OID_OBJ_TYPE = '1.3.6.1.4.1.2.6.190.4.11'
+    OID_SYS_NAME = '1.3.6.1.4.1.2.6.190.4.7'
+    OID_SEVERITY = '1.3.6.1.6.3.1.1.4.1.0'
+
+    TRAP_SEVERITY_MAP = {
+        '1.3.6.1.4.1.2.6.190.1': constants.Severity.CRITICAL,
+        '1.3.6.1.4.1.2.6.190.2': constants.Severity.WARNING,
+        '1.3.6.1.4.1.2.6.190.3': constants.Severity.INFORMATIONAL,
+    }
+
+    SEVERITY_MAP = {"warning": "Warning",
+                    "informational": "Informational",
+                    "error": "Major"
+                    }
+
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+        self.ssh_pool = SSHPool(**kwargs)
+
+    def set_storage_id(self, storage_id):
+        self.storage_id = storage_id
+
+    @staticmethod
+    def handle_split(split_str, split_char, arr_number):
+        split_value = ''
+        if split_str is not None:
+            tmp_value = split_str.split(split_char, 1)
+            if arr_number == 1 and len(tmp_value) > 1:
+                split_value = tmp_value[arr_number].strip()
+            elif arr_number == 0:
+                split_value = tmp_value[arr_number].strip()
+        return split_value
+
+    @staticmethod
+    def parse_alert(context, alert):
+        try:
+            alert_model = dict()
+            alert_name = SSHHandler.handle_split(alert.get(
+                SSHHandler.OID_ERR_ID), ':', 1)
+            error_info = SSHHandler.handle_split(alert.get(
+                SSHHandler.OID_ERR_ID), ':', 0)
+            alert_id = SSHHandler.handle_split(error_info, '=', 1)
+            severity = SSHHandler.TRAP_SEVERITY_MAP.get(
+                alert.get(SSHHandler.OID_SEVERITY),
+                constants.Severity.INFORMATIONAL
+            )
+            alert_model['alert_id'] = str(alert_id)
+            alert_model['alert_name'] = alert_name
+            alert_model['severity'] = severity
+            alert_model['category'] = 'Fault'
+            alert_model['type'] = constants.EventType.EQUIPMENT_ALARM
+            alert_model['sequence_number'] = SSHHandler. \
+                handle_split(alert.get(SSHHandler.OID_SEQ_NUMBER), '=', 1)
+            timestamp = SSHHandler. \
+                handle_split(alert.get(SSHHandler.OID_LAST_TIME), '=', 1)
+            time_type = '%a %b %d %H:%M:%S %Y'
+            occur_time = int(time.mktime(time.strptime(
+                timestamp,
+                time_type)))
+            alert_model['occur_time'] = int(occur_time * 1000)
+            alert_model['description'] = alert_name
+            alert_model['resource_type'] = SSHHandler.handle_split(
+                alert.get(SSHHandler.OID_OBJ_TYPE), '=', 1)
+            alert_model['location'] = SSHHandler.handle_split(alert.get(
+                SSHHandler.OID_SYS_NAME), '=', 1)
+            return alert_model
+        except Exception as e:
+            LOG.error(e)
+            msg = ("Failed to build alert model as some "
```

Comment:
Could this msg be improved to indicate which attribute is missing?

Reply:
changed